contracts/StablecoinConverter.sol

```diff
@@ -108,4 +109,62 @@ contract StablecoinConverter is EpochTokenLocker {
     function tokenIdToAddressMap(uint16 id) public view returns (address) {
         return IdToAddressBiMap.getAddressAt(registeredTokens, id);
     }
-}
\ No newline at end of file
+
+    struct TradeData {
+        address owner;
+        uint volume;
+        uint16 orderId;
+    }
+
+    function submitSolution(
+        uint32 batchIndex,
+        address[] memory owner,  //tradeData is submitted as arrays
+        uint16[] memory orderId,
+        uint128[] memory volume,
+        uint128[] memory prices,
+        uint16[] memory tokenIdForPrice
+    ) public {
+        require(
+            batchIndex == getCurrentStateIndex() - 1,
+            "Solutions are no longer accepted for this batch"
+        );
+        uint len = owner.length;
+        for (uint i = 0; i < len; i++) {
+            Order memory order = orders[owner[i]][orderId[i]];
```

Comment:
Should this be storage so that it can be used below instead of always referring to `orders[owner[i]][orderId[i]]` which is very hard to read.

Reply:
But storage reads are more expensive than storing it once in memory and reading it from there, isn't it?